### PR TITLE
allow symfony 5

### PR DESCRIPTION
**Pull request by Jeroeny**
### Context

Symfony 5 has been released.

### What has been done

- Add `^5.0` to symfony packages in `composer.json`.

### How to test

- Unit tests

### Notes

#SymfonyHackday

### CODE REVIEW

**Comment by tiagobrito** [comment]
@Jeroeny 
You need to create a new class (in `namespace VCR\Event`) to support the Symfony 5 change on the Event class.
 
##### src/VCR/Event.php

```php

<?php

namespace VCR\Event;

// inferior to Symfony/Contracts PHP minimal version
if (class_exists(\Symfony\Contracts\EventDispatcher\Event::class)) {
    class Event extends \Symfony\Contracts\EventDispatcher\Event
    {
    }
} else {
    class Event extends \Symfony\Component\EventDispatcher\Event
    {
    }
}
```

then all the other classes inside `VCR\Event` namespace need to extend from this new `Event` class

**Comment by Jeroeny** [comment]
> @Jeroeny
> You need to create a new class (in `namespace VCR\Event`) to support the Symfony 5 change on the Event class.
> 
> ##### src/VCR/Event.php
> ```
> 
> <?php
> 
> namespace VCR\Event;
> 
> // inferior to Symfony/Contracts PHP minimal version
> if (class_exists(\Symfony\Contracts\EventDispatcher\Event::class)) {
>     class Event extends \Symfony\Contracts\EventDispatcher\Event
>     {
>     }
> } else {
>     class Event extends \Symfony\Component\EventDispatcher\Event
>     {
>     }
> }
> ```
> 
> then all the other classes inside `VCR\Event` namespace need to extend from this new `Event` class

Thanks, I've added it.

File: src/VCR/Event/Event.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
```
Comment (tiagobrito) [inline src/VCR/Event/Event.php]: You can't add this. PHP-VCR supports PHP 5.3, 5.4 and 5.5 and `declare(strict_types=1);` isn't supported on those versions

Comment (Jeroeny) [reply src/VCR/Event/Event.php]: My bad, got left over from my IDE template I think. Removed.

File: src/VCR/Videorecorder.php [hunk 2]
```diff
@@ -101,7 +101,7 @@ public function getEventDispatcher()
      */
     private function dispatch($eventName, Event $event = null)
     {
-        return $this->getEventDispatcher()->dispatch($eventName, $event);
+        return $this->getEventDispatcher()->dispatch($event, $eventName);
```
Comment (derrabus) [inline src/VCR/Videorecorder.php]: This is going to break on Symfony < 4.3

**Comment by pauladams8** [comment]
Could we prioritize symfony 5 compatibility? I would like to use this library but it's causing lots of conflicts in my application.

**Comment by JeroenVanOort** [comment]
As support for Symfony 5 has been added to the 1.5 branch by https://github.com/php-vcr/php-vcr/pull/311 and https://github.com/php-vcr/php-vcr/pull/319, I think it's appropriate to close this PR.

**Pull request closed without merging**